trie/verkle.go

```diff
@@ -239,45 +237,31 @@ func nodeToDBKey(n verkle.VerkleNode) []byte {
 // Commit writes all nodes to the trie's memory database, tracking the internal
 // and external (for account tries) references.
 func (trie *VerkleTrie) Commit(onleaf LeafCallback) (common.Hash, int, error) {
-	flush := make(chan verkle.VerkleNode)
-	flusher := func(n verkle.VerkleNode) {
+	root, ok := trie.root.(*verkle.InternalNode)
+	if !ok {
+		return common.Hash{}, 0, errors.New("unexpected root node type")
+	}
+	nodes, err := root.BatchSerialize()
+	if err != nil {
+		return common.Hash{}, 0, fmt.Errorf("serializing tree nodes: %s", err)
+	}
+
+	for _, node := range nodes {
 		if onleaf != nil {
-			if leaf, isLeaf := n.(*verkle.LeafNode); isLeaf {
-				for i := 0; i < verkle.NodeWidth; i++ {
-					if leaf.Value(i) != nil {
-						comm := n.Commitment().Bytes()
-						onleaf(nil, nil, leaf.Value(i), common.BytesToHash(comm[:]))
+			if leaf, isLeaf := node.Node.(*verkle.LeafNode); isLeaf {
+				for j := 0; j < verkle.NodeWidth; j++ {
+					if leaf.Value(j) != nil {
+						onleaf(nil, nil, leaf.Value(j), common.BytesToHash(node.CommitmentBytes[:]))
 					}
 				}
 			}
 		}
-		flush <- n
-	}
-	go func() {
-		switch root := trie.root.(type) {
-		case *verkle.StatelessNode:
-			root.Flush(flusher)
-		case *verkle.InternalNode:
-			root.Flush(flusher)
-		default:
-			log.Crit("root was not flushed")
-		}
-		close(flush)
-	}()
-	var commitCount int
-	for n := range flush {
-		commitCount += 1
-		value, err := n.Serialize()
-		if err != nil {
-			panic(err)
-		}
-
-		if err := trie.db.DiskDB().Put(nodeToDBKey(n), value); err != nil {
```

Comment:
Note another saving here. We don't do `nodeToDBKey(..)` here, but again leverage work done by the batch serialization since that again did inverses.

The [BatchSerialize(...)](https://github.com/gballet/go-verkle/pull/333/files#diff-050830ac4334170e8d335e1b95519052a3461c5b57a4ca38a0531e35a8f3d388R1229-R1233) has a nice API to use here since it returns a slice of:
```
type SerializedNode struct {
	Node            VerkleNode
	CommitmentBytes [32]byte
	SerializedBytes []byte
}
```
So we receive everything we need to operate at this level (i.e: go-ethereum).